stack/clone_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stack_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/madlambda/spells/assert"
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/stack"
+	"github.com/mineiros-io/terramate/test"
+	"github.com/mineiros-io/terramate/test/sandbox"
+)
+
+func TestStackClone(t *testing.T) {
+	type testcase struct {
+		name    string
+		layout  []string
+		src     string
+		target  string
+		wantErr error
+	}
+
+	testcases := []testcase{
```

Comment:
For now tests are quite simple since we always check that the entire subtrees are identical. For checking for the "almost" identical but with different ID will be tricky =P (also some other scenarios where the stack has other configs inside etc...).